client/payment-details/payment-method/affirm/index.tsx

```diff
@@ -32,20 +34,36 @@ const formatPaymentMethodDetails = ( charge ) => {
 /**
  * Placeholders to display while loading.
  */
-const paymentMethodPlaceholders = {
+const paymentMethodPlaceholders: PaymentMethodDetails = {
 	id: 'id placeholder',
 	name: 'name placeholder',
 	email: 'email placeholder',
 	formattedAddress: 'address placeholder',
 };
 
-const CardDetails = ( { charge = {}, isLoading } ) => {
-	const details =
+/**
+ * Props interface for AffirmDetails component
+ */
+interface AffirmDetailsProps {
+	charge?: Charge;
+	isLoading: boolean;
+}
+
+const AffirmDetails: React.FC< AffirmDetailsProps > = ( {
+	charge,
+	isLoading,
+} ) => {
+	const details: PaymentMethodDetails | typeof paymentMethodPlaceholders =
 		charge && charge.payment_method_details
```

Comment:
```suggestion
		charge && charge.billing_details
```
Should we check for the presence of `billing_details` here? AFAICT, it doesn't look like `payment_method_details` is used inside the `formatPaymentMethodDetails()` function, so I'm wondering what is the purpose of checking for this particular property? 🤔

Reply:
Good point! I’m not sure about this since the implementation is not new and also occurs in other payment method components. I summarized everything for this comment and the below ones in https://github.com/Automattic/woocommerce-payments/pull/8964#issuecomment-2169137429.